OData/src/System.Web.OData/OData/ODataSwaggerConverter.cs

```diff
@@ -0,0 +1,298 @@
+
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Web.Http;
+using Microsoft.OData.Edm;
+using Newtonsoft.Json.Linq;
+
+namespace System.Web.OData
+{
+    /// <summary>
+    /// Represents an <see cref="ODataSwaggerConverter"/> used to converter an Edm model to Swagger model.
+    /// </summary>
+    public class ODataSwaggerConverter
+    {
+        private static readonly Uri DefaultMetadataUri = new Uri("http://localhost");
+        private const string DefaultHost = "default";
+        private const string DefaultbasePath = "/odata";
+
+        /// <summary>
+        /// Gets or sets the metadata Uri in the Swagger model.
+        /// </summary>
+        public Uri MetadataUri { get; set; }
+
+        /// <summary>
+        /// Gets or sets the host in the Swagger model.
+        /// </summary>
+        public string Host { get; set; }
+
+        /// <summary>
+        /// Gets or sets the base path in the Swagger model.
+        /// </summary>
+        public string BasePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Edm model.
+        /// </summary>
+        public IEdmModel EdmModel { get; private set; }
+
+        /// <summary>
+        /// Gets the version of Swagger spec.
+        /// </summary>
+        public virtual Version SwaggerVersion
+        {
+            get
+            {
+                return new Version(2, 0);
+            }
+        }
+
+        /// <summary>
+        /// Gets the Swagger model.
+        /// </summary>
+        public virtual JObject SwaggerModel
+        {
+            get
+            {
+                if (SwaggerDoc == null)
+                {
+                    ConvertToSwaggerModel();
+                }
+
+                Contract.Assert(SwaggerDoc != null);
+                return SwaggerDoc;
+            }
+        }
+
+        /// <summary>
+        /// Gets the document in the Swagger.
+        /// </summary>
+        protected virtual JObject SwaggerDoc { get; set; }
+
+        /// <summary>
+        /// Gets the paths in the Swagger.
+        /// </summary>
+        protected virtual JObject SwaggerPaths { get; set; }
+
+        /// <summary>
+        /// Gets the definitions in the Swagger.
+        /// </summary>
+        protected virtual JObject SwaggerDefinitions { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataSwaggerConverter" /> class.
+        /// </summary>
+        /// <param name="model">The Edm model.</param>
+        public ODataSwaggerConverter(IEdmModel model)
+        {
+            if (model == null)
+            {
+                throw Error.ArgumentNull("model");
+            }
+
+            EdmModel = model;
+            MetadataUri = DefaultMetadataUri;
+            Host = DefaultHost;
+            BasePath = DefaultbasePath;
+        }
+
+        /// <summary>
+        /// Convert the Edm model to Swagger model.
```

Comment:
Converts? Same for verbs below. Since Verbs above are all have suffixes. Just keep consistent, with or without suffix.